server/train/pipe_util.py

```diff
@@ -248,3 +259,214 @@ def merge_model(core_model, dram_model):
     output = layers.Add()([core_model.output, dram_model.output])
     model = Model(inputs=core_model.inputs + dram_model.inputs, outputs=output)
     return model
+
+
+# XGBoost (Gradient Boosting Regressor) Base Model Generation
+class XGBoostRegressionModelGenerationPipeline():
+    """A class used to handle XGBoost Regression Model Incremental Training. This class currently only handles numerical features.
+
+    ...
+
+    Attributes
+    ----------
+    TODO
+
+    Methods
+    -------
+    TODO
+
+    """
+
+    feature_names: List[str]
+    label_names: List[str]
+    model_name: str
+
+    def __init__(self, feature_names_in_order: List[str], label_names_in_order: List[str], save_location: str, model_name: str) -> None:
+        # model data will be generated consistently using the list of feature names and labels (Order does not matter)
+
+        self.feature_names = feature_names_in_order.copy()
+        self.feature_names.sort()
+        self.label_names = label_names_in_order.copy()
+        self.label_names.sort()
+        # allow save_location to be modified
+        self.save_location = save_location
+        self.model_name = model_name
+        self.model_filename = model_name + '.model'
+        self.model_desc = 'model_desc.json'
+
+
+    @staticmethod
+    def _generate_base_model() -> xgb.XGBRegressor:
+        # n_estimators, max_depth, eta (learning rate)
+        return xgb.XGBRegressor(n_estimators=1000, learning_rate=0.1)
```

Comment:
if curious, is there any hyperparaemeter tuning in xgboost package, like `GridSearchCV` and `RandomizedSearchCV` in sklearn?

Reply:
There is hyperparameter tuning because xgboost does use tools from sklearn. If I recall, we need to provide a list of candidate values for key parameters like n_estimators, max_depth, learning rate, etc. While I am not too sure, I think this can be incorporated in the online training. I will look into it. I will include this in the next PR.